dev_guide/builds.adoc

```diff
@@ -385,6 +384,31 @@ sourceStrategy:
 You can also manage environment variables defined in the `*BuildConfig*` with the
 xref:../dev_guide/environment_variables.adoc#overview[`oc set env`] command.
 
+[[adding-secrets-to-source-strategy-build-configs]]
+=== Adding Secrets to Source Strategy Build Configurations
+
+Add a secret to your build configuration so that it can access a private
+repository.
+
+. Create a new {product-title} project.
+
+. xref:../dev_guide/secrets.adoc#dev-guide-secrets[Create a secret] that contains
+credentials for accessing a private repository.
+
+. Create a
+xref:..//architecture/core_concepts/builds_and_image_streams.adoc#source-build[Source-to-Image (S2I)
+build configuration].
+
+. On the build configuration editor page or in the *fromimage* page, set the
+*Source Secret*.
+
```

Comment:
I wonder whether we need document more words about fromimage page because reader may not know where it is